lib/v2/screens/dashboard/wallet/components/transactions/interactor/viewmodels/member_bloc.dart

```diff
@@ -1,20 +1,51 @@
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:hive/hive.dart';
+import 'package:seeds/v2/constants/system_accounts.dart';
+import 'package:seeds/v2/datasource/local/member_model_cache_item.dart';
 import 'package:seeds/v2/domain-shared/page_state.dart';
 import 'package:seeds/v2/screens/dashboard/wallet/components/transactions/interactor/mappers/member_state_mapper.dart';
 import 'package:seeds/v2/screens/dashboard/wallet/components/transactions/interactor/usecases/load_member_data_usecase.dart';
 import 'package:seeds/v2/screens/dashboard/wallet/components/transactions/interactor/viewmodels/member_events.dart';
 import 'package:seeds/v2/screens/dashboard/wallet/components/transactions/interactor/viewmodels/member_state.dart';
 
 
+const cacheExpiryMinutes = 30;
+
 class MemberBloc extends Bloc<MemberEvent, MemberState> {
   MemberBloc(String accountName) : super(MemberState.initial(accountName));
 
   @override
   Stream<MemberState> mapEventToState(MemberEvent event) async* {
     if (event is OnLoadMemberData) {
+      var account = event.account;
+
+      // handle special system accounts
+      if (SystemAccounts.isSystemAccount(account)) {
+        yield state.copyWith(pageState: PageState.success, member: SystemAccounts.getSystemAccount(account)!);
+        return;
+      }
+
       yield state.copyWith(pageState: PageState.loading);
+      var box = await Hive.openBox('members_box');
+
+      // If we have a cached item, use it
+      MemberModelCacheItem? cacheItem = box.get(account);
+      if (cacheItem != null) {
+        yield state.copyWith(pageState: PageState.success, member: cacheItem.member);
+        if (cacheItem.refreshTimeStamp < DateTime.now().millisecondsSinceEpoch) {
+          return;
+        }
+      }
+
+      final result = await LoadMemberDataUseCase().run(account);
 
-      final result = await LoadMemberDataUseCase().run(event.account);
+      // store result in cache
+      if (!result.isError && result.asValue != null) {
```

Comment:
i think we should move this code out of the bloc directly and user mappers to help us do so.

Reply:
I don't think it can be in mappers.

Reply:
@gguijarro-c-chwy Ok I tried moving it into the mapper, and found the issue again...

The neat thing about this cache implementation is that it can yield state multiple times!

1 - item is in cache, yield cached item - this means the item shows immediately on the UX
2 -cached item is stale, load item, then yield new state with new item -> this means the item gets updated.

That was the idea, to be able to yield 2 different results in case of a stale cache entry

List of all possible cases:
A - Cache entry, but it is old - yield cache entry, then later yield fresh item - can't be in mapper as mapper can only return 1 state. 
B - No cache entry - normal loading, and entering into cache. This could be in mapper.
C - Cache entry that's still fresh - show cached item. This could be in mapper.